src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-topic.service.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ApiClient } from './api-client';
+import { TopicModel } from '~/app/ceph/rgw/rgw-topic-list/topic.model';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RgwTopicService extends ApiClient {
+  baseURL = 'api/rgw/topic';
+
+  constructor(private http: HttpClient) {
+    super();
+  }
+
+  list(uid?: string, tenant?: string): Observable<TopicModel[]> {
+    let params = new HttpParams();
+    if (uid) {
+      params = params.append('uid', uid);
+    } else if (tenant) {
+      params = params.append('tenant', tenant);
+    }
```

Comment:
If you send uid, you can not set a tenant?

Should it be?
```suggestion
    if (uid) {
      params = params.append('uid', uid);
   if (tenant) {
      params = params.append('tenant', tenant);
    }
```